application/src/main/java/org/togetherjava/tjbot/commands/SlashCommandAdapter.java

```diff
@@ -154,4 +159,46 @@ public void onSelectionMenu(@NotNull SelectionMenuEvent event, @NotNull List<Str
         return Objects.requireNonNull(componentIdGenerator)
             .generate(new ComponentId(getName(), Arrays.asList(args)), lifespan);
     }
+
+    /**
+     * This method copies the given {@link OptionData} for the given amount of times into a
+     * {@link List}. <br>
+     * This makes all the {@link OptionData OptionData's} optional! Everything else gets exactly
+     * copied.
+     *
+     * @param optionData The {@link OptionData} to copy.
+     * @param amount The amount of times to copy
+     *
+     * @return An unmodifiable {@link List} of the copied {@link OptionData OptionData's}
+     *
+     * @see #varArgOptionsToList(Collection, Function)
+     */
+    @Unmodifiable
+    protected static final @NotNull List<OptionData> generateOptionalVarArgList(
+            final @NotNull OptionData optionData, @Range(from = 1, to = 25) final int amount) {
+
+        OptionData varArgOption = new OptionData(optionData.getType(), optionData.getName(),
+                optionData.getDescription());
+
+        return IntStream.range(0, amount).mapToObj(i -> varArgOption).toList();
+    }
+
+    /**
+     * This method takes a {@link Collection} of {@link OptionMapping OptionMapping's}, these get
+     * mapped using the given {@link Function}
+     *
+     * @param options A {@link Collection} of {@link OptionMapping OptionMapping's}.
+     * @param mapper The mapper {@link Function}
+     * @param <T> The type to map it to.
+     *
+     * @return A modifiable {@link List} of the given type
+     *
+     * @see #generateOptionalVarArgList(OptionData, int)
+     */
+    protected static <T> List<T> varArgOptionsToList(
+            final @NotNull Collection<? extends OptionMapping> options,
+            final @NotNull Function<? super OptionMapping, ? extends T> mapper) {
+
+        return options.stream().map(mapper).collect(Collectors.toCollection(ArrayList::new));
+    }
```

Comment:
> I don't want to keep it specific to RoleSelectCommand since other people should have easy access to this, if one notices the possibility of vararg, they might get ideas for their command with it.

I just dont think that anyone else will want to use it - as of today. It is kinda specific. Maybe the next user will have slightly different needs, for example mixing types or changing names. At which point you would have to completely rework or split the utility anyways.

Its hard to design general purpose utility without having a clear picture of all its users yet. Right now it is just `RoleCommand` that needs it. So I would probably keep it there and elevate it up only if someone else needs it as well (and then we know what the requirements from the other user are).